### PR TITLE
Refactored the fish case and examining fish. New bluespace fish case to fit large fish inside a backpack.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
I've refactored the FISH_SAFE_STORAGE trait into an element, mainly because there were a few problems with how it worked: It wasn't stopping hunger from raising, ditto with the breeding wait, and it offered no healing whatsoever, which I find kind of a bummer. The new element will keep the fish from getting any hungrier or ready to reproduce and will heal them up to 65% of their base health.

Also, I've added a new bluespace fish case as a techweb design, found along with other stuff in the advanced fishing node, though if you want I can move it to a bluespace node. This should make it possible to reasonably store and carry around larger fish.

Examining a fish will no longer give out readings on weight and size if you haven't at least attempted fishing once (you can get to novice level in less than a minute). While examining a fish with apprentice level or higher will also give readings on the general conditions of the fish (is it starving? drowning? has it lost a considerable chunk of health?). The fishing skillchip also gives you these traits.

I've also converted two fish variables into traits, because fish have waaaay too many variables. and gave some topdown shading to the pre-existing fishbox sprite.

## Changelog

:cl:
add: Added a bluespace fish case to the advanced fishing node.
balance: Fish cases will keep a fish from getting hungrier or ready to reproduce, while also healing it up to 65% health.
balance: Examining a fish with zero fishing skill whatsoever won't give a reading on its size and weight. Conversely, examining one with the skill leveled two times will give general information on if it's starving, sick, hungry, or dead.
/:cl:

### CODE REVIEW

**Comment by deathrobotpunch** [comment]
just wondering are the fish gotten from the black market meant to be dead?

**Comment by Ghommie** [comment]
Yes.

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/datums/elements/fish_safe_storage.dm [hunk 1]
```diff
@@ -0,0 +1,53 @@
+///An element that puts in stasis any fish that enters the atom.
+/datum/element/fish_safe_storage
```
[the remark this answers is not in the document]
Comment (Ghommie) [reply code/datums/elements/fish_safe_storage.dm:2]: fewer instances, fewer things to process. It's just a microop.

File: code/datums/elements/fish_safe_storage.dm [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		tracked_fish -= gone
+		gone.exit_stasis()
+
+/datum/element/fish_safe_storage/process(seconds_per_tick)
+	for(var/obj/item/fish/fish as anything in tracked_fish)
+		///Keep delaying hunger and breeding while in stasis, and also heal them.
+		fish.last_feeding += seconds_per_tick SECONDS
+		fish.breeding_wait += seconds_per_tick SECONDS
+		if(fish.health < initial(fish.health) * 0.65)
+			fish.adjust_health(fish.health + 0.75 * seconds_per_tick)
```
[the remark this answers is not in the document]
Comment (Ghommie) [reply code/datums/elements/fish_safe_storage.dm:53]: Likely not many. Almost no one puts fish inside moisture traps or toilets, and fishing is not so popular to have gorgillions or people storing gazillions of fishes inside stasis boxes.

**Comment by Ghommie** [comment]
Drafting because I've realized there's something I've to do.

**Pull request merged**